schema/workflow.json

```diff
@@ -228,20 +228,27 @@
     "error": {
       "type": "object",
       "properties": {
-        "expression": {
+        "name": {
+          "type": "string",
+          "description": "Domain-specific error name, or '*' to indicate all possible errors",
+          "minLength": 1
+        },
+        "code": {
           "type": "string",
-          "description": "JsonPath expression. Evaluated against error data. Is true if returns non empty result"
+          "description": "Error code. Can be used in addition to the name to help runtimes resolve to technical errors/exceptions",
+          "minLength": 1
         },
-        "errorDataFilter": {
-          "$ref": "#/definitions/errordatafilter"
+        "retry": {
```

Comment:
ok good point. i think there are good points for both approaches but i do see the value of decoupling. i will update here in a minute

Reply:
one thing i do want to mention is regarding "all other error are assumed to be transient/retryable"
please note that all errors , caught or retried **have** to be defined. 
we will not have a "catch-all" retry. we cannot issue retries on errors that the users do not think of (are not defined). so that part I will not do in this pr